packages/chain-adapters/src/utxo/UTXOBaseAdapter.ts

```diff
@@ -0,0 +1,159 @@
+    http: bitcoin.api.V1Api //?
+    ws: bitcoin.ws.Client
+  }
+  coinName: string
+  chainId?: CAIP2
+}
+
+/**
+ * Base chain adapter for all UTXO chains. When extending please add your ChainType to the
+ * UTXOChainTypes. For example:
+ *
+ * `export type UTXOChainTypes = ChainTypes.Bitcoin | ChainTypes.Litecoin`
+ */
+export abstract class UTXOBaseAdapter<T extends UTXOChainTypes> implements IChainAdapter<T> {
+  protected chainId: CAIP2
+  protected coinName: string
+  protected readonly providers: {
+    http: bitcoin.api.V1Api
+    ws: bitcoin.ws.Client
+  }
+
+  public static readonly defaultBIP44Params: BIP44Params = {
+    purpose: 84, // segwit native
+    coinType: 0,
+    accountNumber: 0
+  }
+
+  protected constructor(args: ChainAdapterArgs) {
+    this.providers = args.providers
+  }
+
+  /* Abstract Methods */
+
+  abstract subscribeTxs(
+    input: chainAdapters.SubscribeTxsInput,
+    onMessage: (msg: chainAdapters.SubscribeTxsMessage<T>) => void,
+    onError?: (err: chainAdapters.SubscribeError) => void
+  ): Promise<void>
+  abstract unsubscribeTxs(input?: chainAdapters.SubscribeTxsInput): void
+  abstract closeTxs(): void
+  abstract getType(): T
+
+  abstract getTxHistory(
+    input: chainAdapters.TxHistoryInput
+  ): Promise<chainAdapters.TxHistoryResponse<T>>
+
+  abstract buildSendTransaction(
+    tx: chainAdapters.BuildSendTxInput<T>
+  ): Promise<{ txToSign: chainAdapters.ChainTxType<T> }>
+
+  abstract getAddress(input: chainAdapters.GetAddressInput): Promise<string>
+
+  abstract getFeeData(
+    input: Partial<chainAdapters.GetFeeDataInput<T>>
+  ): Promise<chainAdapters.FeeDataEstimate<T>>
+
+  abstract signTransaction(
+    signTxInput: chainAdapters.SignTxInput<chainAdapters.ChainTxType<T>>
+  ): Promise<string>
+
+  /* public methods */
+
+  getCaip2(): CAIP2 {
+    return this.chainId
+  }
+
+  getChainId(): CAIP2 {
+    return this.chainId
+  }
+
+  async getAccount(pubkey: string): Promise<chainAdapters.Account<T>> {
+    if (!pubkey) {
+      return ErrorHandler('UTXOBaseAdapter: pubkey parameter is not defined')
+    }
+
+    try {
+      const caip = await this.getCaip2()
+      const { chain, network } = caip2.fromCAIP2(caip)
+      const { data } = await this.providers.http.getAccount({ pubkey: pubkey })
+
+      const balance = bnOrZero(data.balance).plus(bnOrZero(data.unconfirmedBalance))
+
+      return {
+        balance: balance.toString(),
+        chain: this.getType(),
+        caip2: caip,
+        caip19: caip19.toCAIP19({ chain, network }),
+        chainSpecific: {
+          addresses: data.addresses,
+          nextChangeAddressIndex: data.nextChangeAddressIndex,
+          nextReceiveAddressIndex: data.nextReceiveAddressIndex
+        },
+        pubkey: data.pubkey
+      } as chainAdapters.Account<T>
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async broadcastTransaction(hex: string): Promise<string> {
+    const { data } = await this.providers.http.sendTx({
```

Comment:
this prob needs to to move to concrete class too

Reply:
same comment as above